checker/tests/initialization/TypeFrames5.java

```diff
@@ -0,0 +1,16 @@
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.initialization.qual.UnderInitialization;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class TypeFrames5 {
+    public @Nullable Object f;
+
+    public TypeFrames5(boolean dummy) {
+        @UnderInitialization(TypeFrames5.class) TypeFrames5 a = this;
```

Comment:
I added this test to show that the type of `this` is correct even for non-final classes, which was previously also broken.
Can you think of another case to test?